### PR TITLE
chore: ignore cargo.lock file

### LINKED ISSUE #28: Ignore Cargo.lock file
### Describe this problem

For libraries, cargo.lock shouldn't be checked in git.

### Steps to reproduce

Not required

### Expected behavior

_No response_

### Additional Information

https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html

### PR DESCRIPTION
**Pull request by Rishav1707**
# Which issue does this PR close?

Closes #28 

# What changes are included in this PR?

- Added cargo.lock file in the `.gitignore` file

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@
 # For normal case, output file should not be ignored, and it acts as
 # a signal that our testcase need to be fixed.
 examples/bad-case/simple/select.output
+# Ignoring cargo.lock file
+Cargo.lock
```
Comment (jiacai2050) [inline .gitignore:7]: I'm afraid this is not enough, you also need to delete Cargo.lock from git, 
```
git rm Cargo.lock
```
And remove lock check in our CI
- https://github.com/ceresdb/sqlness/blob/main/.github/workflows/ci.yml#L51
- https://github.com/ceresdb/sqlness/blob/main/.github/workflows/ci.yml#L85

Comment (Rishav1707) [reply .gitignore:7]: > I'm afraid this is not enough, you also need to delete Cargo.lock from git,
> 
> ```
> git rm Cargo.lock
> ```
> 
> And remove lock check in our CI
> 
> * https://github.com/ceresdb/sqlness/blob/main/.github/workflows/ci.yml#L51
> * https://github.com/ceresdb/sqlness/blob/main/.github/workflows/ci.yml#L85

Done

**Comment by jiacai2050** [comment]
Ooops, it seems there are some clippy errors in CI, I guess it's introduced in latest Rust version(rust version 1.67.0 (fc594f156 2023-01-24)), could you fix this in this PR?

**Comment by Rishav1707** [comment]
> Ooops, it seems there are some clippy errors in CI, I guess it's introduced in latest Rust version(rust version 1.67.0 (fc594f156 2023-01-24)), could you fix this in this PR?

But, how to do that?

**Comment by jiacai2050** [comment]
Never mind, I have fixed the clippy issue. 

Usually you can fix with `cargo clippy --fix` command

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**